### PR TITLE
chore(build): Add support for drift detection of generated code

### PR DESCRIPTION
**Pull request by hermanschaaf**
This adds a new workflow that checks that generated code remains the same when go generate is run again. It currently needs to be specifically enabled by adding //check-for-changes to the top of the generator .hcl file.

### CODE REVIEW

**Review by erezrokah — CHANGES_REQUESTED** [review]
🔥 Same as 🚀  Same comment as https://github.com/cloudquery/cq-provider-gcp/pull/468#pullrequestreview-1064896387 regarding paths filters

File: resources/services/teams/teams.go [hunk 1]
```diff
@@ -370,7 +371,7 @@ func Teams() *schema.Table {
 					{
 						Name:        "team_id",
 						Description: "The id of the team",
-						Type:        schema.TypeBigInt,
+						Type:        schema.TypeString,
```
Comment (hermanschaaf) [inline resources/services/teams/teams.go]: @roneli I don't know whether this should be `TypeBigInt` or `TypeString` -- the generator generates String, so I went with that for now.

File: resources/services/teams/teams.go [hunk 2]
```diff
@@ -120,7 +121,7 @@ func Teams() *schema.Table {
 					{
 						Name:        "team_id",
 						Description: "The id of the name",
-						Type:        schema.TypeBigInt,
+						Type:        schema.TypeString,
```
Comment (hermanschaaf) [inline resources/services/teams/teams.go]: @roneli I don't know whether this should be `TypeBigInt` or `TypeString` -- the generator generates String, so I went with that for now.

File: resources/services/teams/teams.go (hunk 1 above)
Comment (hermanschaaf) [reply resources/services/teams/teams.go]: Oh nevermind, the tests failed. Guess it should be TypeBigInt after all

**Comment by hermanschaaf** [comment]
I have now also added the drift detection to all existing files and fixed any drift that had already occurred. To get the script to work also involved bringing the directory structure in line with other providers.

**Review by erezrokah — APPROVED** [review]

**Review by roneli — APPROVED** [review]
LGTM, one minor nit as we moved to packages feels maybe that billing/action_billing.go is redundant and we can do billing/actions etc' wdyt?

**Comment by hermanschaaf** [comment]
@roneli I agree with you, however I think if we change the resource, that would also change the table name. Maybe let's do that in a separate PR if we're going to make a breaking change.

**Pull request merged**